src/frontend/hooks/server/projects.ts

```diff
@@ -9,3 +11,30 @@ export function useProject() {
   const projectContext = useActiveProjectContext();
   return projectContext.project;
 }
+
+export function useAllProjects() {
+  const api = useApi();
+
+  return useQuery({
+    queryFn: async () => await api.listProjects(),
+    queryKey: ['projects'],
+    throwOnError: true,
+  });
+}
+
+export function useCreateProject() {
+  const api = useApi();
+  const queryClient = useQueryClient();
+  const updateProject = useUpdateActiveProjectId();
+
+  return useMutation({
+    mutationKey: ['createProject'],
+    mutationFn: async (name: string) => {
+      return await api.createProject({name});
+    },
+    onSuccess: async data => {
+      updateProject(data);
```

Comment:
wondering if this shouldn't be done here and should be handled by the caller of the hook instead? having this here assumes that a created project should always be immediately switched to, which is accurate for now but seems overly strict/opinionated

Reply:
to more succinctly put it, seems safer to let the caller of the mutation handle side effects and only let the mutation handle mutation-specific things (e.g. interacting with the api)

Reply:
Agreed, this should be handled by the user. Especially in the future when there are multiple project that can be toggled by the user! But for this first QA build, we have decided that if you join a project, you are automatically just part of that project. Ill add a note here that this is not how it should be handled in the future